### ISSUE TITLE
Future Idea: Conditional Scoring

### ISSUE THREAD

**Issue by khannatanmai** [apertium/apertium-anaphora#27]
Give scores to antecedents based on certain conditions of the anaphors, instead of giving the same scores to all antecedents. 

For eg., if reflexive anaphor then a higher weight could be given to the subject.

Basically, can give the user the ability to add such scores.

**Comment by khannatanmai** [comment]
@ftyers to implement this, as we discussed, I'll give the user the ability to define multiple anaphors and name them, and similarly for them to score based on the name of the anaphors, so that they can give different scores to different kinds of anaphors.

**Comment by khannatanmai** [comment]
ARX file modification sample:
`<anaphor n="detpos">`
`<parameter-item has-tags="det pos"/>`
`</anaphor>`
`<anaphor n="verbal">`
`<parameter-item has-tags="v"/>`
`</anaphor>`

`<score n="-1" anaphor="detpos"/>`
`<score n="1" anaphor="verbal"/>`

**Comment by khannatanmai** [comment]
The antecedent tag will also get a anaphor name
`<antecedent n="detpos">
<parameter-item has-tags="n"/>
<parameter-item has-tags="np"/>
`</antecedent>

**Comment by khannatanmai** [comment]
TO add in documentation:
Parameter Types: anaphor, antecedent, delimiter.
Parameter Names: detpos, verbal, etc.

**Comment by khannatanmai** [comment]
NEW IDEA: Instead of just adding types to anaphors, we need to add types to antecedents as well, since different kinds of anaphors may have different needs for antecedents. So, how about I add the ability to define types of parameters and give each of them a name. Each will have its own definition of anaphor, antecedent, and delimiter. For eg.,

```
<section-parameters>
		<def-param n="detpos">
			<anaphor>
				<parameter-item has-tags="det pos"/>
			</anaphor>

			<antecedent>
				<parameter-item has-tags="n"/>
				<parameter-item has-tags="np"/>
			</antecedent>

			<delimiter>
				<parameter-item has-tags="sent"/>
			</delimiter>
		</def-param>


		<def-param n="verbal">
			<anaphor>
				<parameter-item has-tags="v"/>
			</anaphor>

			<antecedent>
				<parameter-item has-tags="n"/>
				<parameter-item has-tags="np"/>
			</antecedent>

			<delimiter>
				<parameter-item has-tags="sent"/>
			</delimiter>
		</def-param>
</section-parameters>
```

This will make it a much more general feature. It is going to take a lot of fundamental changes in the code but it could work. What do you think @ftyers

**Comment by khannatanmai** [comment]
Finally implemented 3fa874c01f7bf687851f014a8d8113afd07e087c

Just going to test it a bit and then close this.

See the arx file in samples/
